aten/src/ATen/native/cuda/Indexing.cu

```diff
@@ -1073,11 +1073,11 @@ void index_reduce_func_cuda_impl(
   // total size of the tensor ignoring dimension `dim`;
   // -the number of index we are choosing, which is the total size
   // of the tensor `index`.
-  ptrdiff_t sliceSize = getSliceSize(self_, dim, index, source_);
-  ptrdiff_t sourceTotalSize = source.numel();
-  int64_t selfReduceDimSize = self_.size(dim);
-  ptrdiff_t numIndex = index.numel();
-  int64_t selfNumel = self_.numel();
+  uint64_t sliceSize = getSliceSize(self_, dim, index, source_);
```

Comment:
Thanks @hongxiayang for the fix! 
In the description you mentioned 1 and 2, and the diff looks mainly fix 1 by using uint64_t which has larger representable range. But I just wondering if we actually should fix `2: On ROCm, std::min -> ::min did not work as expected` instead? (although uint64_t has much larger representable, but in a extreme, in the future if our LLM context length continue to grow, will it to a point that beyond uint64_t representable range?)